beacon_node/network/src/sync/network_context.rs

```diff
@@ -641,8 +726,12 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
             return Some(Err(e));
         }
 
-        if let Some(blocks_result) = entry.get().responses(&self.chain.spec) {
-            entry.remove();
+        if let Some(blocks_result) = entry.get_mut().responses(&self.chain.spec) {
+            if blocks_result.is_ok() {
+                // remove the entry only if it coupled successfully with
+                // no errors
+                entry.remove();
```

Comment:
Is there a retry counter somewhere? A peer should not be able cause a request to hang here forever in an infinite retry loop

Reply:
In this version, I am assuming that there are some good peers that will eventually move the chain forward or all peers will get kicked and the chain will get removed.
Can add a retry counter in another PR.

Reply:
Not all errors result in downscore, so it possible to have infinite loops for chains with a single peer